### PR TITLE
[mongo-cxx-driver] fix static build

### PR DESCRIPTION
**Pull request by sdcb**
note:
Actually, static build is perfectly fine but the cmake always build static and dynamic library togather, in the static environment, dynamic build will fail due link error.
I twisted a patch to the CMakeLists.txt to make dynamic build happy when building static.

### CODE REVIEW

**Comment by alexkaratarakis** [comment]
Thanks!

Building scripts producing both dynamic and static seems to be a recurring pattern. `zlib` does this as well and (at the moment) we just discard the parts we don't need. 
Not producing them at all is better of course!

File: ports/mongo-cxx-driver/portfile.cmake [hunk 1]
```diff
@@ -15,15 +10,22 @@ vcpkg_extract_source_archive(${ARCHIVE})
...
+	set(ENABLE_SHARED ON)
+endif()
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
 	OPTIONS
 		-DLIBBSON_DIR=${CURRENT_INSTALLED_DIR}
 		-DLIBMONGOC_DIR=${CURRENT_INSTALLED_DIR}
 		-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=ON
+		-DENABLE_SHARED=${ENABLE_SHARED}
```
Comment (ras0219-msft) [inline ports/mongo-cxx-driver/portfile.cmake]: Note for the future: this variable already exists and is automatically filled in correctly for every cmake build (`BUILD_SHARED_LIBS`).

Comment (sdcb) [inline ports/mongo-cxx-driver/portfile.cmake]: Thank for you to notice that, I submitted another PR here to fix it: https://github.com/Microsoft/vcpkg/pull/326
